### PR TITLE
view item details

### PR DESCRIPTION
**Pull request by JeffNewcomer**
screen shot 2018-06-29 at 4 37 50 pm
screen shot 2018-06-29 at 4 38 22 pm

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -91,10 +91,20 @@ var app = {
   }
 }
 
+// added these 2 lines of code below so that I can work on the large card. remove later.
```
Comment (thebearingedge) [inline main.js]: You can delete this temporary comment and code.

File: main.js [hunk 2]
```diff
@@ -91,10 +91,20 @@ var app = {
...
+// app.view = 'details'
+// app.details.item = app.catalog.items[1]
+
+// function 1 - create small cards
 function createCard(item) {
   var $card = document.createElement('div')
   $card.classList.add('card')
 
+  // add attribute to root element of the card
+  var a = document.createAttribute('data-item-id')
```
Comment (thebearingedge) [inline main.js]: `$element.setAttribute` creates the `AttributeNode` for you, so you don't need to create `a` here.

File: main.js [hunk 3]
```diff
@@ -148,10 +158,144 @@ function createGridAndHeading(allItems) {
...
+function createFullDetailsCard(item) {
+  var $largeCard = document.createElement('div')
+
+  var $largeCardImageContainer = document.createElement('div')
+  $largeCardImageContainer.classList.add('large-card-image-container')
+  $largeCard.appendChild($largeCardImageContainer)
+
+  var $largeImage = document.createElement('img')
+  $largeCardImageContainer.appendChild($largeImage)
+  // $largeImage.classList.add('card-img-top')
```
Comment (thebearingedge) [inline main.js]: Please delete all of your commented-out code.

File: styles.css [hunk 4]
```diff
@@ -11,7 +11,50 @@ body {
...
+.hidden {
+  display: none;
+}
+
+.lrg-card-div {
+  background-color: white;
+  height: 420px;
+  margin-top: 50px;
+
+  /* border: 5px solid black; */
```
Comment (thebearingedge) [inline styles.css]: Please remove commented-out code.

File: main.js [hunk 5]
```diff
@@ -148,10 +158,144 @@ function createGridAndHeading(allItems) {
...
+  }
 }
-renderGridAndHeader()
+// addHiddenClass('details')
+// addHiddenClass('catalog')
+// do I need to also need to remove 'hidden' from other container(s) when running this function a 2nd time?
+
+// Define a function that takes a view name ('catalog' or 'details'?) and adds a 'hidden' class to all
+// data-view containers (see divs in html) that don't match that view.
+// function addHiddenClass(viewName) {
```
Comment (thebearingedge) [inline main.js]: It's not good to turn in chunks of commented-out code like this.

File: main.js [hunk 6]
```diff
@@ -148,10 +158,144 @@ function createGridAndHeading(allItems) {
...
+
+$catalog.addEventListener('click', displayLargeCard)
+
+function displayLargeCard() {
+  var clickedCard = event.target.closest('.card')
+  var clickedCardItemId = clickedCard.getAttribute('data-item-id')
+  app.view = 'details'
+  var itemObject = getItemObject(clickedCardItemId, app.catalog.items)
+  app.details.item = itemObject
+  console.log(itemObject)
```
Comment (thebearingedge) [inline main.js]: Please remove extra logging.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Great work! You can make a few clean-up changes.

**Comment by JeffNewcomer** [comment]
Thanks.  Made those changes.

**Review by thebearingedge — APPROVED** [review]
👍 Nice!

**Pull request merged**